functions/pupil-prefs/index.js

```diff
@@ -1,19 +1,97 @@
 'use strict'
 
-const uuid = require('uuid/v4')
 const moment = require('moment')
+const sqlService = require('less-tedious')
+const { TYPES } = require('tedious')
+const uuid = require('uuid/v4')
+const winston = require('winston')
+
+const config = require('../config')
+const accessArrangementsTable = '[accessArrangements]'
+const checkTable = '[check]'
+const pupilAccessArrangementsTable = '[pupilAccessArrangements]'
+const pupilFontSizesTable = '[pupilFontSizes]'
+const pupilColourContrastsTable = '[pupilColourContrasts]'
+const pupilTable = '[pupil]'
+const schema = '[mtc_admin]'
 
-module.exports = function (context, prefsMessage) {
-  context.log('pupil-preferences message received', prefsMessage.checkCode)
+winston.level = 'error'
+sqlService.initialise(config)
+
+module.exports = async function (context, pupilPrefsMessage) {
+  const { checkCode, preferences } = pupilPrefsMessage
+  const { fontSizeCode, colourContrastCode } = preferences
+  context.log('pupil-preferences message received', checkCode)

```

Comment:
you should really validate the structure of the message first.  it could have missing properties that breaks code halfway through the function.

Reply:
@GuyHarwood made a push to validate both `checkCode` and the code values (if both do not exist)